### ISSUE TITLE
Token Account == 1 amount logic in mip1 execute sale

### ISSUE THREAD

**Issue by Jac0xb** [magicoss/m2#1]
https://github.com/magicoss/m2/blob/4764e3dcfb7f516251f3196af545436273d1cb98/programs/m2/src/mip1_ins/mip1_execute_sale_v2.rs#L363

Shouldn't this try to borrow the token account receipt post-cpi transfer and make sure its amount == 1 and seller token account is == 0 instead of checking the seller's token account?

**Comment by madergaser** [comment]
Yeah, I think that is the most correct way to check it, but at this point of the contract execution, I think we have already successfully transferred the token, and since (IIRC) pNFTs are always supply 1, we can avoid the extra serialization/deserialization and check the outdated copy of seller's token account.
